hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/SortPhysicalRule.java

```diff
@@ -19,35 +19,112 @@
 import com.hazelcast.jet.sql.impl.opt.OptUtils;
 import com.hazelcast.jet.sql.impl.opt.logical.SortLogicalRel;
 import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelCollationTraitDef;
+import org.apache.calcite.rel.RelFieldCollation;
 import org.apache.calcite.rel.RelNode;
-import org.apache.calcite.rel.convert.ConverterRule;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import static com.hazelcast.jet.sql.impl.opt.JetConventions.LOGICAL;
-import static com.hazelcast.jet.sql.impl.opt.JetConventions.PHYSICAL;
+import static java.util.Collections.singletonList;
 
-final class SortPhysicalRule extends ConverterRule {
+final class SortPhysicalRule extends RelOptRule {
 
     static final RelOptRule INSTANCE = new SortPhysicalRule();
 
     private SortPhysicalRule() {
         super(
-                SortLogicalRel.class, LOGICAL, PHYSICAL,
+                operand(SortLogicalRel.class, LOGICAL, some(operand(RelNode.class, any()))),
                 SortPhysicalRule.class.getSimpleName()
         );
     }
 
     @Override
-    public RelNode convert(RelNode rel) {
-        SortLogicalRel logicalLimit = (SortLogicalRel) rel;
+    public void onMatch(RelOptRuleCall call) {
+        SortLogicalRel logicalSort = call.rel(0);
+
+        List<RelNode> transforms = toTransforms(logicalSort);
+        for (RelNode transform : transforms) {
+            call.transformTo(transform);
+        }
+    }
+
+    private static List<RelNode> toTransforms(SortLogicalRel logicalSort) {
+        List<RelNode> sortTransforms = new ArrayList<>(1);
+        List<RelNode> nonSortTransforms = new ArrayList<>(1);
+        for (RelNode physicalInput : OptUtils.extractPhysicalRelsFromAllSubsets(logicalSort.getInput())) {
+            boolean requiresSort = requiresLocalSort(
+                    logicalSort.getCollation(),
+                    physicalInput.getTraitSet().getTrait(RelCollationTraitDef.INSTANCE)
+            );
+            if (requiresSort) {
```

Comment:
I have some serious doubts regarding correctness of the below:
- if `requiresSort` we always add same rel (converted `logicalSort`) to `sortTransforms`
- if `logicalSort.offset != null || logicalSort.fetch != null` we create `SortPhysicalRel` twice, seems unnecessary; moreover commenting it out didn't fail any test (at least could not find one)